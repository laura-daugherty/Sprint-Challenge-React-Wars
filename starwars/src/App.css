.App {
  text-align: center;
}
.Header {
  color: black;
  text-shadow: 1px 1px 5px #fff;
  font-size: 3rem;
  font-family: 'Geostar Fill', cursive;
}
/* write your parent styles in here for your App.js */
