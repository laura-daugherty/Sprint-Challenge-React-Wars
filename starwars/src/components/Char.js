import React from 'react'

const Char = props => {
  if (!props.character) {
    return (
      <div 
        className='ui segment'
        style={ {  
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          backgroundColor: "black",
          color: "goldenrod",
          maxWidth: "700px",
          height: "200px",
        }}
    >
      <h2 
        className='ui heading'
        style={ { 
          fontFamily: "'Geostar Fill', cursive",
          textShadow: "1px 1px 1px #fff"
        }}>
          Press Button Above to Fetch Your First Character
      </h2>
    </div>
    )
  } else {
    return (
      <div className='ui segment'
        style={ {  
          backgroundColor: "black",
          color: "goldenrod",
          width: "700px",
        }}
      >
        <h2 className='charName'
          style={ {
            fontFamily: "'Geostar Fill', cursive",
            textShadow: "1px 1px 1px #fff"
          }}>
            {props.character.name}
        </h2>
        <h3>Gender: {props.character.gender}</h3>
        <h3>Height: {props.character.height}</h3>
        <h3>Birth Year: {props.character.birth_year}</h3>
      </div>
    );
  };
};

export default Char;
