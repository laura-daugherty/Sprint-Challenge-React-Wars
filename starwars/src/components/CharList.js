import React from 'react'
import './StarWars.css'

import Char from './Char.js';

import { Loader } from "semantic-ui-react";

function CharList(props) {
  console.log("charList props", props)

  if (!props.character) {
    return (
      <div 
        className="ui active inverted dimmer">
        <Loader className="ui large text loader">
          Click the Button
        </Loader>
      </div>
    )
  } else {
    return (
      <Char character={props.character} clickHandler={changeCharacter}/>
    );
  }
}

export default CharList;