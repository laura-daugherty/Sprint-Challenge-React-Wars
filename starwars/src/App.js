import React, { useState, useEffect } from 'react';
import axios from "axios"
import './App.css';
import CharList from './components/CharList.js'

import "semantic-ui-css/semantic.min.css";

function App() {
  // constructor() {
  //   super();
  //   this.state = {
  //     starwarsChars: []
  //   };
  // }

  // componentDidMount() {
  //   this.getCharacters('https://swapi.co/api/people/');
  // }

  // getCharacters = URL => {
  //   // feel free to research what this code is doing.
  //   // At a high level we are calling an API to fetch some starwars data from the open web.
  //   // We then take that data and resolve it our state.
  //   fetch(URL)
  //     .then(res => {
  //       return res.json();
  //     })
  //     .then(data => {
  //       this.setState({ starwarsChars: data.results });
  //     })
  //     .catch(err => {
  //       throw new Error(err);
  //     });
  // };

  const [data, setData] = useState("");
  const [error, setError] = useState("");

  useEffect(() => {
    console.log("useEffect fetching Data")
    axios
      .get("https://swapi.co/api/people/")
      .then(result => setData(result.data.results))
      .catch(err => setError("THERES AN ERROR!"));
  }, [] );

  console.log({data})
  
  if(error) {
    return (
      <h2
        className="ui red block header centered"
      >
        {error}
      </h2>
    )
  } else {
    return (
      <div className='App'>
        <h1 className='Header'>React Wars</h1>
        <CharList data={data} />
      </div>
    );
  }
}

export default App;
