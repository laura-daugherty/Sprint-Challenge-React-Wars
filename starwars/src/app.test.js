import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'

import App from './App';
import CharList from "./components/CharList"
import Char from "./components/Char"

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('<App />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<CharList />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<CharList />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<Char />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<Char />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});