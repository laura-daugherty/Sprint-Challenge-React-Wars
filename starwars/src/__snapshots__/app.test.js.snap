// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="App"
  style={
    Object {
      "alignItems": "center",
      "backgroundImage": "url(sw-bg.jpg)",
      "display": "flex",
      "flexDirection": "column",
      "fontFamily": "Orbitron, sans-serif",
      "height": "-webkit-fill-available",
      "margin": "0",
      "padding": "0",
      "textAlign": "center",
    }
  }
>
  <header>
    <h1
      className="ui header"
      style={
        Object {
          "color": "black",
          "fontFamily": "Geostar Fill, cursive",
          "fontSize": "3rem",
          "textShadow": "1px 1px 5px #fff",
        }
      }
    >
      React Wars
    </h1>
    <div
      className="ui active inverted dimmer"
    >
      <div
        className="ui text loader ui large text loader"
      >
        Fetching Your Characters
      </div>
    </div>
  </header>
  <div
    className="ui segment"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "color": "goldenrod",
        "display": "flex",
        "height": "200px",
        "justifyContent": "center",
        "maxWidth": "700px",
      }
    }
  >
    <h2
      className="ui heading"
      style={
        Object {
          "fontFamily": "'Geostar Fill', cursive",
          "textShadow": "1px 1px 1px #fff",
        }
      }
    >
      Press Button Above to Fetch Your First Character
    </h2>
  </div>
</div>
`;

exports[`<Char /> should match snapshot 1`] = `
<div
  className="ui segment"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "color": "goldenrod",
      "display": "flex",
      "height": "200px",
      "justifyContent": "center",
      "maxWidth": "700px",
    }
  }
>
  <h2
    className="ui heading"
    style={
      Object {
        "fontFamily": "'Geostar Fill', cursive",
        "textShadow": "1px 1px 1px #fff",
      }
    }
  >
    Press Button Above to Fetch Your First Character
  </h2>
</div>
`;

exports[`<CharList /> should match snapshot 1`] = `
<div
  className="ui active inverted dimmer"
>
  <div
    className="ui text loader ui large text loader"
  >
    Click the Button
  </div>
</div>
`;
